src/RelationsAns.lagda

```diff
@@ -19,6 +19,21 @@ open even
 open odd
 \end{code}
 
+* Irreflexivity
+
+\begin{code}
+open import Logic using (⊥-elim; excluded-middle; ¬_; _⊎_; inj₁; inj₂)
+
+<-irrefl : ∀ (n : ℕ) → excluded-middle → ¬ ( n < n )
+<-irrefl zero _ ()
+<-irrefl (suc n) em with em {A = suc n < suc n}
+<-irrefl (suc n) em | inj₁ sn<sn = ⊥-elim (<-irrefl n em (helper n sn<sn))
+  where
+    helper : ∀ (n : ℕ) → suc n < suc n → n < n
+    helper n (s<s n<n) = n<n
+<-irrefl (suc n) em | inj₂ ¬sn<sn = ¬sn<sn
+\end{code}
+
```

Comment:
This was adapted from[ this first version.](https://gist.github.com/jonaprieto/80a8bdd6b200a8bb8fab865ced43bae7)